network/ClientNetworking.cpp

```diff
@@ -235,10 +279,15 @@ bool ClientNetworking::ConnectToLocalHostServer(
 }
 
 void ClientNetworking::DisconnectFromServer() {
-    if (Connected())
+    bool is_open(false);
+
+    { // Create a scope for the mutex
+        boost::mutex::scoped_lock lock(m_mutex);
```

Comment:
presumably the other mutex lock is related to cases like this, though some comment about why this is necessary might be helpful...